lib/Crypto/Crypt.php

```diff
@@ -97,6 +109,7 @@ public function __construct(ILogger $logger, IUserSession $userSession, IConfig
 		$this->config = $config;
 		$this->l = $l;
 		$this->supportedKeyFormats = ['hash', 'password'];
+		$this->useLegacyEncoding = $this->config->getSystemValue('encryption.use_legacy_encoding', false);
```

Comment:
The `useLegacyEncoding` isn't guaranteed to have a boolean value here.

Reply:
'true'/'false'

Reply:
> 'yes'/'no'

why? we have many config parameters having true|false.

Reply:
true|false

I took a look in core with already existing boolean config values. maybe a hint :
`if ($includeVersion || (bool) $systemConfig->getValue('version.hide', false) === false) {`
`\OC::$server->getConfig()->getSystemValue('show_server_hostname', false) !== true,`
`\OC::$server->getConfig()->getSystemValue('use_relative_domain_name', false) === true`

Reply:
I just set to false if it is not strictly equal to true.